Tests/MapboxMapsTests/MapView/Integration Tests/OfflineManagerIntegrationTests.swift

```diff
@@ -0,0 +1,158 @@
+                                                      averageBytesPerSecond: nil)!
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        tileRegionLoadOptions = nil
+    }
+
+    // MARK: Test Cases
+
+    internal func testProgressAndCompletionBlocksBaseCase() {
+        /// Expectations to be fulfilled
+        let downloadInProgress = XCTestExpectation(description: "Downloading offline tiles in progress")
+        let completionBlockReached = XCTestExpectation(description: "Checks that completion block closure has been reached")
+
+        /// Perform the download
+        TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                               loadOptions: tileRegionLoadOptions!) { _ in
+            DispatchQueue.main.async {
+                downloadInProgress.fulfill()
+                downloadInProgress.assertForOverFulfill = false
+            }
+        } completion: { result in
+            switch result {
+            case .success(let region):
+                completionBlockReached.fulfill()
+            case .failure(let error):
+                XCTFail("Download failed with error: \(error)")
+            }
+        }
+
+        let expectations = [downloadInProgress, completionBlockReached]
+        wait(for: expectations, timeout: 5.0)
+    }
+
+    internal func testProgressCanBeCancelled() {
+
+        /// Expectations to be fulfilled
+        let downloadInProgress = XCTestExpectation(description: "Downloading offline tiles in progress")
+        let downloadWasCancelled = XCTestExpectation(description: "Checks a cancel function was reached and that the download was canceled")
+
+        /// Perform the download
+        let download = TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                                              loadOptions: tileRegionLoadOptions!) { _ in
+            DispatchQueue.main.async {
+                downloadInProgress.fulfill()
+            }
+        } completion: { _ in }
+
+        /// This guarantees that after 3 seconds of a download in progress, we will force a cancel
+        DispatchQueue.main.asyncAfter(deadline: .now() + 3.0) {
+            self.cancelDownload(download: download, expectation: downloadWasCancelled)
+        }
+
+        let expectations = [downloadInProgress, downloadWasCancelled]
+        wait(for: expectations, timeout: 5.0)
+    }
+
+    internal func testOfflineRegionCanBeDeleted() {
+
+        /// Expectations to be fulfilled
+        let downloadWasDeleted = XCTestExpectation(description: "Downloaded offline tiles were deleted")
+
+        /// Perform the download
+        TileStore.getInstance().loadTileRegion(forId: tileRegionId,
+                                               loadOptions: tileRegionLoadOptions!) { _ in } completion: { _ in }
+
+        TileStore.getInstance().removeTileRegion(forId: self.tileRegionId)
```

Comment:
I swear I left a comment here before. We should the tile region after the load has completed.

Reply:
you might have said this at another place